packages/x-charts/src/ChartsText/ChartsText.tsx

```diff
@@ -24,51 +24,45 @@ function ChartsText(props: ChartsTextProps) {
 
   const { angle, textAnchor, dominantBaseline, ...style } = styleProps ?? {};
 
-  const wordsByLines = React.useMemo(
-    () => getWordsByLines({ style, needsComputation: text.includes('\n'), text }),
-    [style, text],
-  );
+  const lines = text.split('\n');
 
-  let startDy: number;
+  let startDy: string;
   switch (dominantBaseline) {
     case 'hanging':
-      startDy = 0;
+      startDy = '0em';
       break;
     case 'central':
-      startDy = ((wordsByLines.length - 1) / 2) * -wordsByLines[0].height;
+      startDy = `-${(lines.length - 1) / 2}em`;
+      break;
+    case 'text-top':
+      startDy = '0em';
+      break;
+    case 'text-bottom':
+      startDy = `-${lines.length - 0.5}em`;
       break;
     default:
-      startDy = (wordsByLines.length - 1) * -wordsByLines[0].height;
+      startDy = `-${lines.length - 1}em`;
       break;
   }
```

Comment:
I'd rather use `lh` instead of `em` so we can respect `line-height`, but it doesn't work properly in Safari <= 18.3: https://github.com/mui/mui-x/pull/16709#issuecomment-2717217365